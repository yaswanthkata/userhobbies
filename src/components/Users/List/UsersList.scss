@import "../../../styles/variables";
.users-list {
    list-style-type: none;
    height: 400px;
    padding-top:10px; 
    overflow: auto;
    li {
        padding: 10px;
        overflow: auto;
        img {
          float: left;
          margin: 0 15px 0 0;
        }
      }
      
      li:hover {
        background: $color-border-light;
        cursor: pointer;
      }
      li.selected {
        background:$color-border-dark;
      }
      .user-title{
          text-align: start;
          padding: 10px;
      }
  }
 
  @media screen and (max-width: 600px){
    .users-list {
      height: 200px;
    }
  }