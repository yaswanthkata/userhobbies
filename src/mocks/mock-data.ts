import { Hobby } from "./../types";
import { User } from "./../types";

export const hobbies: Hobby[] = [
  { id: 1, name: "Swimming", passion: 0, year: 2008, userId: 1 },
  { id: 2, name: "Cycling", passion: 1, year: 2009, userId: 1 }
];
export const users: User[] = [
  { id: 1, name: "Yaswanth kata", thumbnail: "" },
  { id: 2, name: "Pulkit", thumbnail: "" }
];
