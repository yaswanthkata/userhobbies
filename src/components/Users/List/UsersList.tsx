import React from "react";
import { User } from "../../../types";
import "./UsersList.scss";

export interface IUsersListProps {
  users: User[];
  selectedUserId?: number;
  selectUser(user: User): void;
}
export const UsersList = ({ users, selectedUserId, selectUser }: IUsersListProps) => {
  const onUserSelectionChanged = (user: User) => {
    selectUser(user);
  };

  return (
    <ul className="users-list">
      {users.map((user: User) => (
        <li
          key={user.id}
          onClick={() => onUserSelectionChanged(user)}
          className={user.id === selectedUserId ? "selected" : ""}
        >
          <div>
            <img src={"https://i.pravatar.cc/50?img=" + user.id} alt="avatar" />
            <h3 className="user-title"> {user.name}</h3>
          </div>
        </li>
      ))}
    </ul>
  );
};

export default UsersList;
