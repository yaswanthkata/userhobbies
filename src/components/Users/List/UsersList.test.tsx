import React from 'react';
import { configure, mount, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as mockData from "../../../mocks/mock-data";
import UsersList, { IUsersListProps } from './UsersList';

configure({ adapter: new Adapter() });


const props: IUsersListProps = {
    users: mockData.users,
    selectUser: jest.fn()
}

let componentWrapper: ShallowWrapper<React.Component>;

beforeEach(() => {
    componentWrapper = shallow(<UsersList {...props} />);
});


it('should render users list component correctly', () => {
    expect(componentWrapper).toMatchSnapshot();
});

it('should render one <ul> element', () => {
    expect(componentWrapper.find('ul')).toHaveLength(1);
});

it("should render  a list of users", () => {
    const userListItems = componentWrapper.find('li').length;

    expect(userListItems).toEqual(2);
});
