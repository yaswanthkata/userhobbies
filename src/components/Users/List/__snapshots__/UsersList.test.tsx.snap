// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render users list component correctly 1`] = `ShallowWrapper {}`;
